tests/js/common/shell/shell-figures-rocksdb.js

```diff
@@ -0,0 +1,424 @@
+        assertEqual(2, indexes.length);
+        assertEqual("primary", indexes[0].type);
+        assertEqual(0, indexes[0].id);
+        assertEqual(100, indexes[0].count);
+        assertEqual("rocksdb-ttl", indexes[1].type);
+        assertEqual(idxs[0].id, cn + '/' + indexes[1].id);
+        assertEqual(0, indexes[1].count);
+      } finally {
+        db._drop(cn);
+      }
+    },
+    
+    testDetailedTtl: function () {
+      let c = db._create(cn, { numberOfShards: 4 });
+      try {
+        let idxs = [];
+        idxs.push(c.ensureIndex({ type: "ttl", fields: ["value1"], expireAfter: 100000 }));
+        let dt = (new Date()).toISOString();
+        for (let i = 0; i < 100; ++i) {
+          c.insert({ value1: dt, value2: "test" + i });
+        }
+        let figures = c.figures(true).engine;
+        assertEqual(100, figures.documents);
+
+        let indexes = figures.indexes;
+        assertEqual(2, indexes.length);
+        assertEqual("primary", indexes[0].type);
+        assertEqual(0, indexes[0].id);
+        assertEqual(100, indexes[0].count);
+        assertEqual("rocksdb-ttl", indexes[1].type);
+        assertEqual(idxs[0].id, cn + '/' + indexes[1].id);
+        assertEqual(100, indexes[1].count);
+      } finally {
+        db._drop(cn);
+      }
+    },
+    
+    testDetailedGeoInvalid: function () {
+      let c = db._create(cn, { numberOfShards: 4 });
+      try {
+        let idxs = [];
+        idxs.push(c.ensureIndex({ type: "geo", fields: ["lat", "lon"] }));
+        for (let i = 0; i < 100; ++i) {
+          c.insert({ lat: "piff", lon: "paff" });
+        }
+        let figures = c.figures(true).engine;
+        assertEqual(100, figures.documents);
+
+        let indexes = figures.indexes;
+        assertEqual(2, indexes.length);
+        assertEqual("primary", indexes[0].type);
+        assertEqual(0, indexes[0].id);
+        assertEqual(100, indexes[0].count);
+        assertEqual("geo", indexes[1].type);
+        assertEqual(idxs[0].id, cn + '/' + indexes[1].id);
+        assertEqual(0, indexes[1].count);
+      } finally {
+        db._drop(cn);
+      }
+    },
+    
+    testDetailedGeo: function () {
+      let c = db._create(cn, { numberOfShards: 4 });
+      try {
+        let idxs = [];
+        idxs.push(c.ensureIndex({ type: "geo", fields: ["lat", "lon"] }));
+        for (let i = 0; i < 100; ++i) {
+          c.insert({ lat: 50 - i, lon: 50 - i });
+        }
+        let figures = c.figures(true).engine;
+        assertEqual(100, figures.documents);
+
+        let indexes = figures.indexes;
+        assertEqual(2, indexes.length);
+        assertEqual("primary", indexes[0].type);
+        assertEqual(0, indexes[0].id);
+        assertEqual(100, indexes[0].count);
+        assertEqual("geo", indexes[1].type);
+        assertEqual(idxs[0].id, cn + '/' + indexes[1].id);
+        assertEqual(100, indexes[1].count);
+      } finally {
+        db._drop(cn);
+      }
+    },
+    
+  };
+}
+
```

Comment:
Why omit the smart edge collection test here?

Reply:
Because smart edge collections are unsupported for the use with detailed figures in 3.6, in the sense of that no extra details will be returned and that the already existing figures are off.